python/lsst/ctrl/bps/wms/panda/cmd_line_embedder.py

```diff
@@ -0,0 +1,94 @@
+import os
+
+
+class CommandLineEmbedder:
+    """
+    Class embeds static (constant across a task) values into the pipeline execution command line
+    and resolves submission side environment variables
+
+    Parameters
+    ----------
+    config : `~lsst.ctrl.bps.BPSConfig`
+            BPS configuration that includes the list of dynamic (uniques per job)
+            and submission side resolved variables
+
+    """
+    def __init__(self, config):
+        self.leave_placeholder_params = config.get("placeholderParams", ['qgraphNodeId', 'qgraphId'])
+        self.submit_side_resolved = config.get("submitSideResolvedParams", ['USER'])
+
+    def replace_static_parameters(self, cmd_line, lazy_vars):
+        """ Substitutes the lazy parameters in the command line which are static,
+        the same for every job in the workflow and could be defined once. This function offloads the edge
+        node processing and number of parameters transferred together with job
+
+
+        Parameters
+        ----------
+        cmd_line: `str` command line to be processed
+        lazy_vars: `dict` of lazy variables and its values
+
+        Returns
+        -------
+        processed command line
+        """
+
+        for param_name, param_val in lazy_vars.items():
+            if param_name not in self.leave_placeholder_params:
+                cmd_line = cmd_line.replace('{'+param_name+'}', param_val)
+        return cmd_line
+
+    def resolve_submission_side_env_vars(self, cmd_line):
+        """ Substitutes the lazy parameters in the command line which are defined and resolved
+         on the submission side
+
+        Parameters
+        ----------
+        cmd_line: `str` command line to be processed
+
+        Returns
+        -------
+        processed command line
+        """
+
+        for param in self.submit_side_resolved:
+            cmd_line = cmd_line.replace('<ENV:'+param+'>', os.getenv(param))
+        return cmd_line
+
+    def attach_pseudo_file_params(self, lazy_vars):
+        """ Adds the parameters needed to finalize creation of a pseudo file
+
+        Parameters
+        ----------
+        lazy_vars: `dict` of values of to be substituted
+
+        Returns
+        -------
+        pseudo input file name suffix
+        """
+
+        file_suffix = ""
+        for item in self.leave_placeholder_params:
+            file_suffix += '+'+item+':'+lazy_vars.get(item, '')
+        return file_suffix
+
+    def substitute_command_line(self, bps_file_name, cmd_line, lazy_vars, job_name):
+        """ Preprocesses the command line leaving for the egde node evaluation only parameters which
+        are job / environment dependent
+
+        Parameters
+        ----------
+        bps_file_name: `str` pseudo input file name proposed by BPS

```

Comment:
What is this?  bps doesn't propose any pseudo input filenames.  Is it the GenericWorkflowJob's name?  No that's below.  So not sure what this is.